internal/app/server/server.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 Envoyproxy Authors
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package server
+
+import (
+	"context"
+	"net"
+
+	"github.com/envoyproxy/xds-relay/internal/pkg/log"
+
+	api "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	"github.com/envoyproxy/go-control-plane/pkg/cache"
+	gcp "github.com/envoyproxy/go-control-plane/pkg/server"
+	"google.golang.org/grpc"
+)
+
+// Run instantiates a running gRPC server for accepting incoming xDS-based
+// requests.
+func Run() {
+	logger := log.New().Sugar()
+
+	// Cursory implementation of go-control-plane's server.
+	snapshotCache := cache.NewSnapshotCache(false, cache.IDHash{}, nil)
+	gcpServer := gcp.NewServer(context.Background(), snapshotCache, nil)
```

Comment:
Shouldn't we pass in a WithCancel context so that we can shutdown via SIGINT?

Reply:
Let me update the PR to include `WithCancel`.

This is a very cursory implementation of the server. Only meant to place it to unblock some parallel work streams. i.e., We won't want to be using the snapshot cache in the final implementation either.